src/middleware/AuthMiddleware.ts

```diff
@@ -6,10 +6,21 @@ const authHandler = new AuthHandler();
 
 export const isAdmin = asyncHandler(
   async (req: Request<{}, {}, {}>, res, next: NextFunction) => {
-    const decoded = await authHandler.authenticate(req.headers.authorization);
+    const decoded = await authHandler.authenticateWithJwt(
+      req.headers.authorization
+    );
 
     res.locals.authenticated = true;
     res.locals.username = decoded.payload.username;
     next();
   }
 );
+
+export const hasValidApiKey = asyncHandler(
+  async (req: Request<{}, {}, {}>, res, next: NextFunction) => {
+    await authHandler.authenticateWithApiKey(
+      req.headers["x-api-key"] as string
```

Comment:
Type assertions should be avoided at all cost.
```suggestion
      req.headers["x-api-key"].toString()
```